tests/integration.py

```diff
@@ -307,6 +312,50 @@ def test_custom_predictors(self):
         prog_server.stop()
         prog_server.start()
 
+    def test_custom_estimators(self):
+        # Restart server with model
+        prog_server.stop()
+
+        #define the custom model
+        ball = LinearThrownObject(thrower_height=1.5, throwing_speed=20)
+        x_guess = ball.StateContainer({'x': 1.75, 'v': 35})
+        kf = KalmanFilter(ball, x_guess)
+        prog_server.start(models ={'ball':ball}, port=9883, state_estimators={'kf':kf})
+        ball_session = prog_client.Session('ball', port=9883, state_est='kf')
+
+        # time step (s)
+        dt = 0.01 
+        x = ball.initialize()
+        # Initial input 
+        u = ball.InputContainer({})  
+
+        # Iterate forward 1 second and compare
+        x = ball.next_state(x, u, 1) 
+        ball_session.send_data(time=1, x=x['x'])
+
+        t, x_s = ball_session.get_state()
+
+        # To check if the output state is multivariate normal distribution
+        self.assertIsInstance(x_s, MultivariateNormalDist)
+      
+        # Setup Particle Filter
+        pf = ParticleFilter(ball, x_guess)
+        prog_server.stop()
+        prog_server.start(models ={'ball':ball}, port=9883, state_estimators={'pf': pf, 'kf':kf})
+        ball_session = prog_client.Session('ball', port=9883, state_est='pf')
+        
+        # Iterate forward 1 second and compare
+        x = ball.next_state(x, u, 1) 
+        ball_session.send_data(time=1, x=x['x'])
+        
+        t, x_s = ball_session.get_state()
+
+        # Ensure that PF output is unweighted samples
+        self.assertIsInstance(x_s, UnweightedSamples)
+       
```

Comment:
Could we also add a test for invalid extra state estimators format. Something like:

`
with self.assertRaises(ERROR):
     # state_estimators not a dictionary
     prog_server.start(models ={'ball':ball}, port=9883, state_estimators=45) 
`

Where error is replaced with the actual error raised. This might not work (e.g., if the error isn't raised because the server is in a different process). In which case you would have to do something like this:

`
# state_estimators not a dictionary
prog_server.start(models ={'ball':ball}, port=9883, state_estimators=45) 
with self.assertRaises(ERROR):  # Won't connect
     ball_session = prog_client.Session('ball', port=9883)
`

Reply:
I raised an exception when the state_estimator is in an invalid format. The test worked.